### DISCUSSION TITLE
Ricoh MP C3004ex (ArchLinux) - Need help

### DISCUSSION THREAD

**Discussion by Ka-Mau** [alexpevzner/sane-airscan#303] [Show and tell]
At first I thought it was working like clockwork. But later I got several error messages.
But first info about what I did:
Avahi is normally inactive, so:
1. avahi activated
2. scanimage -L
3. copied output 1:1 into escl.conf
4. avahi disabled
This worked at first, but now it doesn't. Most of the time it ends with the error:
- Document feeder empty
Sometimes it works when I enable avahi, sometimes not. Can anyone here help me to use the Ricoh reliably via sane?
I have the following questions:
What needs to be entered in escl.config? I tried:
```
# device `airscan:e0:RICOH MP C3004ex [XXXXXXXXX]' is a eSCL RICOH MP C3004ex [XXXXXXXXX] ip=192.168.75.77
# "Ricoh_MP_C3004ex" = http://192.168.75.77/eSCL, eSCL
# device `airscan:e0:RICOH MP C3004ex [58387908B62E]' is an eSCL RICOH MP C3004ex [58387908B62E] ip=192.168.75.77
# device http://192.168.74.77:80 "RICOH MP C3004ex [58387908B62E]"
# device airscan:e0:192.168.75.77:80
```

Or is there an error and how can I help to solve it? A mac for debugging purposes is available.

**Comment by Ka-Mau** [comment]
I could probably solve the problem myself by using airscan.conf instead of escl.conf.
```
"Ricoh MP C3004ex" = http://192.168.75.77/eSCL
```
But this time I am careful and don't claim that it "works like a charm".

**Comment by alexpevzner** [comment]
You probably have two backends enabled: sane_airscan and sane_escl

Both will require either working Avahi or static configuration. Configuration files are different and their format too

So the interesting question, which of them works and which doesn't
